File: oop/src/token.ts

export enum TokenType {
    Illegal,
    Simple,
    BulkType,
    ArrayType,
    Len,
    Bulk,
    RetCar,
    NewL,
    Eof,
};

export type Token = {
    type: TokenType;
    literal: string | null;
};

File: client/src/token.rs

#[derive(Debug, Clone, PartialEq, Eq)]
pub enum Token {
    Illegal,
    BulkType,
    ArrayType,
    Retcar,
    Newl,
    Eof,
    Len(std::rc::Rc<str>),
    Bulk(std::rc::Rc<str>),
    Simple(std::rc::Rc<str>),
    Error(std::rc::Rc<str>),
}
